core/matrix/dense.cpp

```diff
@@ -266,8 +266,20 @@ void Dense<ValueType>::inv_scale_impl(const LinOp *alpha)
         GKO_ASSERT_EQUAL_COLS(this, alpha);
     }
     auto exec = this->get_executor();
-    exec->run(dense::make_inv_scale(
-        make_temporary_conversion<ValueType>(alpha).get(), this));
+    // if alpha is real (convertible to real) and ValueType complex
+    if (dynamic_cast<const ConvertibleTo<Dense<>> *>(alpha) &&
```

Comment:
Are the two branches still necessary with the templated kernels? The only difference is choosing the correct conversion of alpha. Perhaps that could be moved before the kernel run, s.t. only one kernel call is necessary.

Reply:
yes, since we use the same piece of code for real and complex ValueType, we need something that compiles for settings, even if it is never taken for real ValueType.